agent/utils.py

```diff
@@ -101,27 +101,42 @@ def _get_target_from_url(message: m.Message) -> Target | None:
             and parsed_url.netloc.split(":")[-1] != ""
         ):
             port = int(parsed_url.netloc.split(":")[-1])
-    args_port = _get_port(message)
+    args_port = _get_port(message, schema)
     port = port or SCHEME_TO_PORT.get(schema) or args_port
     target = Target(host=domain_name, scheme=schema, port=port)
     return target
 
 
-def prepare_target(message: m.Message) -> tuple[str, int]:
+def prepare_targets(message: m.Message) -> list[Target]:
     """Prepare targets based on type, if a domain name is provided, port and protocol are collected
     from the config."""
     if (host := message.data.get("host")) is not None:
-        port = _get_port(message)
-        return host, port
+        scheme = _get_scheme(message)
+        port = _get_port(message, scheme)
+        try:
+            mask = int(message.data.get("mask", None))
```

Comment:
if mask is None this will lead to TypeError .
can you explain what does this code try to check please.

Reply:
The code basically limits the CIDR range that the agent can scan to 255 hosts.